docs/training-run/index.txt

```diff
@@ -4,106 +4,259 @@
+at the top level, which lists the file paths to be omitted from the model definition. The
+``.detignore`` file uses the same syntax as `.gitignore <https://git-scm.com/docs/gitignore>`__.
+Byte-compiled Python files, including ``.pyc`` files and ``__pycache__`` directories, are always
+ignored.
 
 ********************
- Pause and Activate
+ Pre-training Setup
 ********************
 
-An important feature of Determined is the ability to have trials stop running and then start again
-later without losing any training progress. The scheduler might choose to stop running a trial to
-allow a trial from another experiment to run, but a user can also manually pause an experiment at
-any time, which causes all of its trials to stop.
+Trials are created to train the model. The :ref:`Hyperparameter Tuning <hyperparameter-tuning>`
+searcher specified in the experiment configuration file defines a set of hyperparameter
+configurations. Each set corresponds to a single trial.
+
+After the context and experiment configuration reach the master, the experiment waits for the
+scheduler to assign slots. If there are insufficient idle slots available, the master automatically
+scales up the resource pools.
+
+When a trial is ready to run, the master communicates with the agent, or :ref:`distributed training
+<multi-gpu-training>` agents, which create(s) containers that have the configured environment and
+training code. A set of default container images applicable to many deep learning tasks is provided,
+but you can also specify a :ref:`custom image <custom-docker-images>`. If the specified container
+images do not exist locally, the trial container fetches the images from the registry.
+
+After starting the containers, each trial runs the ``startup-hook.sh`` script in the context
+directory.
+
+The pre-training activity can incur a delay before each trial begins training but typically only
+takes a few seconds.
+
+********************
+ Pause and Activate
+********************
 
-Checkpointing is essential to this ability. After a trial is set to be stopped, it takes a
-checkpoint at the next available opportunity (i.e., once its current workload finishes running) and
-then stops running, freeing up the slots it was using. When it resumes running, either because more
-slots become available in the cluster or because a user activates the experiment, it loads the saved
-checkpoint, allowing it to continue training from the same state it had before.
+Trials can be paused and reactivated without losing training progress. The scheduler can stop
+running a trial to allow a trial from another experiment to run, but you can also manually pause an
+experiment at any time, which causes all of its trials to stop. After a trial is set to stop, it
+takes a checkpoint at the next available opportunity, such as after the current workload completes,
+and stops running. This frees the slots used by the trial. When the trial resumes running, because
```

Comment:
running a trial to allow a trial from another experiment to run, but you can also manually pause an
experiment at any time, ~which causes all of its trials to stop. After a trial is set to stop, it
takes a checkpoint at the next available opportunity, such as after the current workload completes,
and stops running.~ which pauses all trials in the experiment.

Reply:
Done